src/cli/main.rs

```diff
@@ -341,12 +361,31 @@ fn format(opt: opt::Opt) -> Result<i32> {
                     let tx = tx.clone();
                     let opt = opt.clone();
 
+                    let should_skip_format = match &opt.stdin_filepath {
+                        Some(filepath) => {
+                            let ignore = get_ignore(
+                                filepath.parent().expect("cannot get parent directory"),
+                                opt.search_parent_directories,
+                            );
```

Comment:
I haven't tested this, but could `.context("failed to parse ignore file")?` work here?

Reply:
Yep, just tested. That works too. Updated.